### PR TITLE
Feature/readme

### PR DESCRIPTION
**Pull request by marco-tranzatto**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,8 +1,73 @@
...
+	git clone git@github.com:ethz-asl/orb_slam_2_catkin.git # cloning over SSH (github account needed)
+	# git clone https://github.com/ethz-asl/orb_slam_2_catkin.git # cloning over HTTPS (no github account needed)
+	```
+- [image_undistort](https://github.com/ethz-asl/image_undistort)
+	```
+	cd ~/catkin_workspace/src
+	git clone git@github.com:ethz-asl/image_undistort.git # cloning over SSH (github account needed)
+	# git clone https://github.com/ethz-asl/image_undistort.git # cloning over HTTPS (no github account needed) 
+	```
+- [pangoli_catkin](https://github.com/ethz-asl/pangolin_catkin)
```
Comment (alexmillane) [inline README.md]: "pangoli" -> "pangolin"

I think that pangolin is actually an rpg package with the link being:

https://github.com/uzh-rpg/pangolin_catkin

File: README.md [hunk 2]
```diff
@@ -1,8 +1,73 @@
 # orb_slam_2_ros
-This package integrates orb_slam_2 into ROS in what I believe to be a more user friendly way than what is offered by the underlying library found here:
 
-https://github.com/raulmur/ORB_SLAM2
+## Overview
 
-Depends on my catkinized version of the orb_slam_2 library:
+This package integrates orb_slam_2 into ROS in a more user friendly way than what is offered by the original library [ORB_SLAM2](https://github.com/raulmur/ORB_SLAM2)
```
Comment (alexmillane) [inline README.md]: I would leave the qualifier in here, "in what I believe to be a more user friendly way".

File: README.md [hunk 3]
```diff
@@ -1,8 +1,73 @@
...
+By building the higher level package orb_slam_2_ros you will even compile its dependencies. This includes the compilation of orb_slam_2_catkin, which will **automatically** install the original library ORB_SLAM2 in your system, meaning you **do not need** to do anything else to then use ORB_SLAM2.
+
+## Usage
+Before launching the node with an example dataset you need to:
+ - Download a dataset you can use to run this package from [EUROC](https://projects.asl.ethz.ch/datasets/doku.php?id=kmavvisualinertialdatasets) (for example the one called "Machine Hall 001") (containing rosbag to play and camera calibration);
+ - Download and extract the [ORB_SLAM2 vocabulary](https://github.com/raulmur/ORB_SLAM2/blob/master/Vocabulary/ORBvoc.txt.tar.gz).
+
+Run the main node with
+
+    roslaunch orb_slam_2_ros run_orb_slam_2.launch vocabulary_file_path:=<PATH_TO_EXTRACTED_ORB_SLAM2_VOCABULARY>
```
Comment (alexmillane) [inline README.md]: This is great! So this actually runs?!

File: README.md [hunk 4]
```diff
@@ -1,8 +1,73 @@
...
 
-https://github.com/ethz-asl/orb_slam_2_catkin
+**Keywords:** ROS, ORB_SLAM2, SLAM
+
+### License
+
+The source code is released under a [TODO @alexmillane insert license](put_link_here).
+
+**Author(s): Alex Millane
+Maintainer: Alex Millane
```
Comment (alexmillane) [inline README.md]: Feel free to add yourself to these lists. By writing you're as much an author/maintainer as I am.

Comment (alexmillane) [reply README.md]: Only if you actually want to of course.

File: README.md (hunk 1 above)
Comment (marco-tranzatto) [reply README.md]: good catch!

File: README.md (hunk 4 above)
Comment (marco-tranzatto) [reply README.md]: I don't think it's fair now. I have just written this readme file ;)

File: README.md (hunk 3 above)
Comment (marco-tranzatto) [reply README.md]: actyally it's not. I need to open a PR and merge my fork to have it running with Euroc, so for now I will just remove this part.

**Comment by marco-tranzatto** [comment]
@alexmillane Since it's not working yet with Euroc I deleted the "Usage" section for now, but asap I'll open a PR from my fork. For now let's just update this first part of the readme file.

File: README.md (hunk 4 above)
Comment (marco-tranzatto) [reply README.md]: also here you need a license :D

File: README.md (hunk 3 above)
Comment (alexmillane) [reply README.md]: OK great.

File: README.md (hunk 4 above)
Comment (alexmillane) [reply README.md]: Added. I have just added the same license as orb_slam.

**Comment by alexmillane** [comment]
@marco-tranzatto Sounds good man.

**Pull request merged**